src/praisonai/tests/unit/test_decorator_simple.py

```
@@ -0,0 +1,102 @@
+
+import sys
+import os
+
+# Add the praisonai-agents module to path
+sys.path.insert(0, os.path.join(os.path.dirname(__file__), 'src', 'praisonai-agents'))
+
+def test_improved_decorator():
+    """Test the improved decorator with context management."""
+    print("🧪 Testing Improved Decorator with Context Management")
+    print("=" * 55)
+    
```

Pylint (3.3.7)

[convention] Trailing whitespace

(C0303)